helper/cert/source_gen.go

```diff
@@ -61,6 +62,12 @@ type GenSource struct {
 	SecretsCache  informerv1.SecretInformer
 	LeaderElector *leader.LeaderElector
 
+	// UpdateCancel and StopUpdate are used to control and sync the goroutine
+	// that continuously attempts to update the K8s Secret with the new
+	// certificate
+	UpdateCancel context.CancelFunc
+	StopUpdate   chan struct{}
+
```

Comment:
You can look into [oklog/run](https://github.com/oklog/run) to do this kind of synchronization.

Reply:
I misread this! It doesn't look like you're spawning multiple goroutines. It's more like handling the lifecycle for a single goroutine that can live past the `Certificate()` call.